src/3rdparty/openttd_social_integration_api/openttd_social_integration_api_v1.h

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023 OpenTTD project
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files
+ * (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* Although all the source-files created by OpenTTD are licensed under the
+ * GPL-v2, this file is an exception. This file is part of the API for
+ * social integration plugins, and licensed under MIT, to allow for non-free
+ * implementations.
+ */
+
+/** @file v1.h Version 1 definition of the OpenTTD Social Integration Plugin API. */
+
+#ifndef OPENTTD_SOCIAL_INTEGRATION_API_V1_H
+#define OPENTTD_SOCIAL_INTEGRATION_API_V1_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Pointers supplied by the plugin for OpenTTD to use. */
+struct OpenTTD_SocialIntegration_v1_PluginInfo {
+	/**
+	 * The Social Platform this plugin is for.
+	 *
+	 * UTF-8, nul-terminated. The plugin is and remains the owner of the memory.
+	 *
+	 * As there can only be one plugin active for each Social Platform, this
+	 * value is used to determine which plugin to use.
+	 *
+	 * Values are like:
+	 * - Steam
+	 * - Discord
+	 * - GOG Galaxy
+	 * - ...
+	 */
+	const char *social_platform;
```

Comment:
Maybe we should set up a wiki page that is the definitive source of social platform names, so when two people are working on XYZ they will both call it XYZ instead of one calling in Xyz. As that would defeat the whole preventing of duplicates. And by not putting it in the source code directly, later additions can still be easily managed.
Also maybe check for duplicate social platform names case insensitively?

Reply:
Solved the last part. First part, sure, we can do that. Leaving open as I have yet to do that :)